spmd/tensor/__init__.py

```diff
@@ -168,7 +169,11 @@ def replicate_module_params_buffers(m: nn.Module, mesh: DeviceMesh) -> None:
 
     # register input_fn as module forward pre hook
     if input_fn is not None:
-        module.register_forward_pre_hook(lambda _, inputs: input_fn(inputs, device_mesh))  # type: ignore[misc]
+        dim = kwargs.get("input_shard_dim", None)
```

Comment:
what's the reason we need this? shouldn't this goes to the make_input_shard/replicate functions directly?

Reply:
The function itself is stateless, the dim is not stored there.

Reply:
I see, but we have a default argument for dim in make_input_shard? should we put the -1 there?

also, maybe we can use functools.partial to generate a partial function that already have the state?